nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/factory/ops/NDBaseTest.java

```diff
@@ -0,0 +1,923 @@
+    public void testAssign() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.eye(3);
+        INDArray y = Nd4j.createUninitialized(x.shape());
+        INDArray z = base.assign(x, y);
+        assertEquals(x, y);
+        assertEquals(z, y);
+
+        y = null;
+        z = base.assign(x, y); //test crashes.
+    }
+
+    @Test
+    public void testConcat() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = Nd4j.ones(DataType.DOUBLE, 3, 3);
+
+        INDArray z = base.concat(new INDArray[]{x, y}, 0);
+        assertArrayEquals(z.shape(), new long[]{6, 3});
+
+        z = base.concat(new INDArray[]{x, y}, 1);
+        assertArrayEquals(z.shape(), new long[]{3, 6});
+    }
+
+    @Test
+    public void testCumprod() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.linspace(DataType.DOUBLE, 1.0, 1.0, 9).reshape(3, 3);
+        INDArray y = base.cumprod(x, false, false, 0);
+        INDArray y_exp = Nd4j.createFromArray(new double[][]{{1.0, 2.0, 3.0}, {4.0, 10.0, 18.0}, {28.0, 80.0, 162.0}});
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+
+        y = base.cumprod(x, false, false, 1);
+        y_exp = Nd4j.createFromArray(new double[][]{{1.0, 2.0, 6.0}, {4.0, 20.0, 120.0}, {7.0, 56.0, 504.0}});
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+
+    }
+
+    @Test
+    public void testCumsum() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.linspace(DataType.DOUBLE, 1.0, 1.0, 9).reshape(3, 3);
+        INDArray y = base.cumsum(x, false, false, 0);
+        INDArray y_exp = Nd4j.createFromArray(new double[][]{{1.0, 2.0, 3.0}, {5.0, 7.0, 9.0}, {12.0, 15.0, 18.0}});
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+
+        y = base.cumsum(x, false, false, 1);
+        y_exp = Nd4j.createFromArray(new double[][]{{1.0, 3.0, 6.0}, {4.0, 9.0, 15.0}, {7.0, 15.0, 24.0}});
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+    }
+
+    @Test
+    public void testDot() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.linspace(DataType.DOUBLE, 1.0, 1.0, 3);
+        INDArray y = base.dot(x, x, 0);
+        INDArray y_exp = Nd4j.scalar(14.0);
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+    }
+
+    @Test
+    public void testDynamicpartition() {
+        //Try to execute the sample in the code dcumentation:
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.linspace(DataType.DOUBLE, 1.0, 1.0, 5);
+        int numPartitions = 2;
+        int[] partitions = new int[]{1, 0, 0, 1, 0};
+        INDArray y = base.dynamicPartition(x, partitions, numPartitions);
+        //TODO: crashes here. Op needs fixing.
+
+    }
+
+    @Test
+    public void testDynamicStitch() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.linspace(DataType.DOUBLE, 1.0, 1.0, 9).reshape(3, 3);
+        INDArray y = base.dynamicStitch(new INDArray[]{x, x}, 0);
+        //TODO: crashes here. Op needs fixing.
```

Comment:
Bad constructor, as previously flagged. Both input and indices need to be INDArrays

Reply:
The +1 here just means I pasted this comment into the test. Will need to resolve the constructor.